src/deprecated.jl

```diff
@@ -0,0 +1,80 @@
+#
+# Deprecated Quadrature things
+const AbstractQuadratureProblem = AbstractIntegralProblem
+const AbstractQuadratureAlgorithm = AbstractIntegralAlgorithm
+const AbstractQuadratureSolution = AbstractIntegralSolution
+
+# Deprecated High Level things
+# All downstream uses need to be removed before removing
+
+const DEAlgorithm = AbstractDEAlgorithm
+const SciMLAlgorithm = AbstractSciMLAlgorithm
+const DEProblem = AbstractDEProblem
+const DEAlgorithm = AbstractDEAlgorithm
+const DESolution = AbstractSciMLSolution
+const SciMLSolution = AbstractSciMLSolution
+
+# Deprecated operator interface
+#=
+const AbstractDiffEqOperator = AbstractSciMLOperator
+const AbstractDiffEqLinearOperator = AbstractSciMLLinearOperator
+const AbstractDiffEqCompositeOperator = AbstractSciMLOperator
+
+const DiffEqScaledOperator = ScaledOperator
+function DiffEqScaledOperator(args...; kwargs...)
+    @warn "SciMLBase.DiffEqScaledOperator is deprecated.
+    Use SciMLOperators.ScaledOperator instead"
+
+    ScaledOperator(args...; kwargs...)
+end
+
+const FactorizedDiffEqArrayOperator = InvertedOperator
+function FactorizedDiffEqArrayOperator(args...; kwargs...)
+    @warn "SciMLBase.FactorizedDiffEqArrayOperator is deprecated.
+    Use SciMLOperators.InvertedOperator instead"
+
+    InvertedOperator(args...; kwargs...)
+end
+
+const DiffEqIdentity = IdentityOperator
+function DiffEqIdentity(u)
+    @warn "SciMLBase.DiffEqIdentity is deprecated.
+    Use SciMLOperators.IdentityOperator instead"
+
+    IdentityOperator{size(u, 1)}()
+end
+
+const DiffEqScalar = SciMLOperators.ScalarOperator
+function DiffEqScalar(args...; kwargs...)
```

Comment:
How does this work if this is happening? Add a downstream test onto LinearSolve.jl

Reply:
DiffEqOperators is not being deprecated in this PR. These lines are all commented out. I am only adding support for `SciMLOperators.jl` by moving `AbstractSciMLOperator` and many of the traits to `SciMLOperators` and making `DiffEqOperators` a subtype. That way, all the stuff in `SciMLOperators` will be supported in downstream packages.
 Read more here: https://github.com/SciML/SciMLOperators.jl/issues/142

Reply:
downstream test for what - i dont follow

Reply:
> and making DiffEqOperators a subtype

I don't see that. I just see a bunch of comments.

Reply:
fair. the comments have been removed. apologies for the mess and thanks for pointing to the issue. I've removed unnecessary comments for this PR.